src/Logger_new.java

```diff
@@ -0,0 +1,340 @@
+        if (isDebugEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void info(final Object msg) {
+        if (isInfoEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void warn(final Object msg) {
+        if (isWarnEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void error(final Object msg) {
+        error(msg, null);
+    }
+
+    public void error(final Object msg, final Throwable ex) {
+        if (isErrorEnabled()) {
+            synchronized (this) {
+                if (!_pbar.isEmpty()) {
+                    clearPogress();
+                    _logWriter.println(msg);
+
+                    if (ex != null) {
+                        ex.printStackTrace(_logWriter);
+                    }
+
+                    printProgress();
+                } else {
+                    _logWriter.println(msg);
+
+                    if (ex != null) {
+                        ex.printStackTrace(_logWriter);
+                    }
+                }
+
+                _logWriter.flush();
+            }
+        }
+    }
+
+    public ProgressBar createProgressBar(final Object message, final int total) {
+        return createProgressBar(message, total, false);
+    }
+
+    public ProgressBar createProgressBar(final Object message, final int total,
+            final boolean closeOnComplete) {
+        if (!isProgressBarSupported() || (total == 0)) {
+            return new DummyProgressBar();
+        }
+
+        synchronized (this) {
+            final PBar pbar = new PBar(message, total, closeOnComplete);
+            _pbar.add(pbar);
+            pbar.start();
+            return pbar;
+        }
+    }
+
+    public boolean isDebugEnabled() {
+        return getLevel() == Level.DEBUG;
+    }
+
+    public boolean isInfoEnabled() {
+        return getLevel().ordinal() <= Level.INFO.ordinal();
+    }
+
+    public boolean isWarnEnabled() {
+        return getLevel().ordinal() <= Level.WARNING.ordinal();
+    }
+
+    public boolean isErrorEnabled() {
+        return getLevel().ordinal() <= Level.ERROR.ordinal();
+    }
+
+    private synchronized void log(final Object msg) {
+        if (!_pbar.isEmpty()) {
+            clearPogress();
+            _logWriter.println(msg);
+            printProgress();
+        } else {
+            _logWriter.println(msg);
+        }
+
+        _logWriter.flush();
+    }
+
+    // @GuardedBy("this")
+    private void clearPogress() {
+        _logWriter.print('\r');
+        for (final PBar b : _pbar) {
+            for (int i = 0; i < b._lastMessageLen; i++) {
+                _logWriter.print(' ');
+            }
+            _logWriter.print(' ');
+        }
+        _logWriter.print('\r');
+    }
+
+    // @GuardedBy("this")
+    private void printProgress() {
+        for (final PBar b : _pbar) {
+            b.printMessage();
+            _logWriter.print(' ');
+        }
+        _logWriter.flush();
+    }
+
+    public static enum Level {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this redundant "static" qualifier. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS2786)